src/pages/AddressInfoPage/index.tsx

```diff
@@ -57,99 +59,139 @@ type ParamTypes = {
   id: string
 }
 
+type FetchedEntity = 'balance' | 'txNumber' | 'assets' | 'transactions'
+
 const TransactionInfoPage = () => {
   const theme = useTheme()
   const { id } = useParams<ParamTypes>()
   const { client, setSnackbarMessage, networkType } = useGlobalContext()
+  const isAppVisible = usePageVisibility()
 
   const [addressBalance, setAddressBalance] = useState<AddressBalance>()
   const [addressTransactionNumber, setAddressTransactionNumber] = useState<number>()
   const [addressAssets, setAddressAssets] = useState<AddressAssetsResult>()
   const [addressTransactions, setAddressTransactions] = useState<Transaction[]>()
+  const [addressMempoolTransactions, setAddressMempoolTransactions] = useState<MempoolTransaction[]>()
   const [addressLatestActivity, setAddressLatestActivity] = useState<number>()
+  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
+  const [exportModalShown, setExportModalShown] = useState(false)
+  const pageNumber = usePageNumber()
 
-  const [loadings, setLoadings] = useState({
+  const knownPendingTxs = useRef<MempoolTransaction[]>([])
+
+  const [loadings, setLoadings] = useState<{ [key in FetchedEntity]: boolean }>({
     balance: true,
     txNumber: true,
     assets: true,
     transactions: true
   })
 
-  const [addressWorth, setAddressWorth] = useState<number | undefined>(undefined)
-
-  const [exportModalShown, setExportModalShown] = useState(false)
+  const fetchTransactionNumber = useCallback(async () => {
```

Comment:
@nop33 If you want a little challenge: dry those fetch calls bellow without overcomplexifying typings :))